cosmoz-treenode-navigator.js

```diff
@@ -162,108 +163,44 @@
-					level = Object.keys(children).map(function (childKey) {
-						var child = children[childKey];
-						return {
-							id: child.key,
-							name: child[this.comparisonProperty],
-							path: child.pathLocator,
-							children: child[this.childProperty]
-						};
-					}, this);
-				} else {
-					pathSegment = children;
-				}
-			}, this);
-
-			return this._sortItOut(level);
+			var n = tree.getNodeByPathLocator(pathLocator),
+				level = tree.getChildren(n) ? _objectValues(tree.getChildren(n)) : n;
+			return this._sortNodes(this._normalizeNodes(_objectValues(level)));
 		},
 		/**
-		 * Returns an Array of nodes on a given path.
+		 * Normalizes and returns an Array of nodes
+		 * with the properties name, path, sectionName, children
 		 */
-		_getNodesOnPath: function (pl, nodes) {
-			var path = pl.split(this.separatorSign),
-				pathSegment = nodes;
-				
-			return path.map(function (nodeKey) {
-				var node = pathSegment[nodeKey],
-					children = node[this.childProperty];
-				node['key'] = nodeKey;
-				if (node && children !== undefined && Object.keys(children).length > 0) {
-					pathSegment = children;
-				}
-				return node;
+		_normalizeNodes: function (nodes) { 
+			return nodes.map(function (node) {
+				var path = node.pathLocator || node.path;
+				return {
+					name: node[this.comparisonProperty],
+					path: path,
+					sectionName: this.tree.getPathString(path, this.comparisonProperty),
+					children: this.tree.getChildren(node)
+				};
 			}, this);
 		},
 		/**
 		 * Returns a node based on a given path locator.
+		 * If pathLocator is empty or not defined, null gets returned.
 		 */
-		_getNode: function (pl, nodes) {
-			if (!pl) {
+		_getNode: function (pathLocator) {
+			if (!pathLocator) {
 				return null;
 			}
-			var pathArray = pl.split(this.separatorSign),
-				pathSegment = nodes,
-				node;
-			
-			pathArray.some(function (path) {
-				node = pathSegment[path];
-
-				if (node === undefined) {
-					console.error('Path does not exist.', pathArray, path, nodes);
-					return true;
-				}
-				var children = node[this.childProperty];
-				if (children !== undefined && Object.keys(children).length > 0) {
-					pathSegment = children;
-				}
-			}, this);
-
-			return node;
+			return this.tree.getNodeByPathLocator(pathLocator);
 		},
-		_getTreePathParts: function (value, data) {
-			if (value === null) {
-				value = '';
-			}
-			var path = value.split(this.separatorSign),
-				parts = [],
-				newpath = [];
-
-			path.some(function (part) {
-				newpath.push(part);
-				var newPathString = newpath.join(this.separatorSign),
-					node = this._getNode(newPathString, data);
-				if (!node) {
-					return true;
-				}
-				node.path = newPathString;
-				parts.push(node);
-			}, this);
-
-			return parts;
+		_getTreePathParts: function (pathLocator, tree) {
+			if (!tree || !pathLocator) {
+				return [];
+			};
+			return this._normalizeNodes(tree.getPath(pathLocator));
```

Comment:
Why do the nodes in the path need to be normalized?
Can't we refactor away `_getTreePathParts()` completely?